utils/db_utils.py

```diff
@@ -424,100 +430,119 @@ def query_trajectories(start_date: str, end_date: str, tz: str):
 
 
 
-def add_user_stats(user_data):
+def add_user_stats(user_data, batch_size=5):
+    time_format = 'YYYY-MM-DD HH:mm:ss'
     with ect.Timer() as total_timer:
-
-        for user in user_data:
-            user_uuid = UUID(user['user_id'])
-
-            # Stage 1: Count total trips
-            with ect.Timer() as stage1_timer:
-                total_trips = esta.TimeSeries.get_aggregate_time_series().find_entries_count(
-                    key_list=["analysis/confirmed_trip"],
-                    extra_query_list=[{'user_id': user_uuid}]
-                )
-                user['total_trips'] = total_trips
-            esdsq.store_dashboard_time(
-                "admin/db_utils/add_user_stats/count_total_trips",
-                stage1_timer
-            )
-
-            # Stage 2: Count labeled trips
-            with ect.Timer() as stage2_timer:
-                labeled_trips = esta.TimeSeries.get_aggregate_time_series().find_entries_count(
-                    key_list=["analysis/confirmed_trip"],
-                    extra_query_list=[{'user_id': user_uuid}, {'data.user_input': {'$ne': {}}}]
-                )
-                user['labeled_trips'] = labeled_trips
-            esdsq.store_dashboard_time(
-                "admin/db_utils/add_user_stats/count_labeled_trips",
-                stage2_timer
-            )
-
-            # Stage 3: Retrieve user profile data
-            with ect.Timer() as stage3_timer:
-                profile_data = edb.get_profile_db().find_one({'user_id': user_uuid})
-                user['platform'] = profile_data.get('curr_platform')
-                user['manufacturer'] = profile_data.get('manufacturer')
-                user['app_version'] = profile_data.get('client_app_version')
-                user['os_version'] = profile_data.get('client_os_version')
-                user['phone_lang'] = profile_data.get('phone_lang')
-            esdsq.store_dashboard_time(
-                "admin/db_utils/add_user_stats/retrieve_user_profile_data",
-                stage3_timer
-            )
-
-            if total_trips > 0:
-                # Stage 4: Get first trip timestamp
-                with ect.Timer() as stage4_timer:
-                    time_format = 'YYYY-MM-DD HH:mm:ss'
-                    ts = esta.TimeSeries.get_time_series(user_uuid)
-                    start_ts = ts.get_first_value_for_field(
-                        key='analysis/confirmed_trip',
-                        field='data.end_ts',
-                        sort_order=pymongo.ASCENDING
+        # Stage 1: Define process_user
+        with ect.Timer() as stage1_timer:
+            def process_user(user):
```

Comment:
why are we defining this within the timer? We want to measure execution time, not definition time.

Reply:
Seems to be a dup/unnecessary as you suggest